Source/WebCore/Modules/push-api/PushEvent.h

```diff
@@ -26,6 +26,8 @@
 #pragma once
 
 #include "ExtendableEvent.h"
+#include "Notification.h"
```

Comment:
Notification.h is wrapped in ENABLE(NOTIFICATIONS), but this file has no feature guards whatsoever. How do you mean for the added code to be guarded?
(PlayStation is currently `!ENABLE(NOTIFICATIONS)` and cannot build.)

Reply:
I looked at this with @rkirsling and my take is that a lot this new `PushEvent` stuff needs to be wrapped in `ENABLE(DECLARATIVE_WEB_PUSH)`; it won’t be terribly difficult, but it’s not trivial. Should re-land with that done

Reply:
That was missed since we have no such EWS coverage.
The additions to PushEvent.h/cpp/idl that are conditioned on that need to be guarded.
Without the failing build in front of me I'm not sure I could make meaningful progress on that - Is there a link to a build outside our EWS infrastructure that I could see?

Reply:
You'd be able to look at the PlayStation post-merge builders, though it'd probably be easier to temporarily set ENABLE_NOTIFICATIONS to 0 in PlatformEnable.h. If it had just been a matter of missing a spot or two, I'd already have a fix landed, but the edit to the existing `PushEvent` constructor is where things get less trivial.

Reply:
Here's one possible fix:
https://gist.github.com/rkirsling/0f4c2ed4aa514142f75f7274d70b0744

Reply:
> easier to temporarily set ENABLE_NOTIFICATIONS to 0 in PlatformEnable.h...

This may not be as trivial as you suggest for a platform that has all of the related features enabled as well!
But I'll try.

Reply:
True, it appears that the actual way to do this on Mac is to add
```
#undef ENABLE_DECLARATIVE_WEB_PUSH
#undef ENABLE_NOTIFICATION_EVENT
#undef ENABLE_NOTIFICATIONS
```
above all of the `#define`s in PlatformEnable.h.

It would also be possible to do a best guess without building anything, which might lead you to make the exact fix patch above. A best guess would have been enough to avoid a revert here.

Reply:
IMHO, that fix isn't right.

Because the problem isn't that Playstation doesn't have DECLARATIVE_WEB_PUSH off...  It's that it has NOTIFICATIONS off.

And on the topic of best guesses - There's a reason we have EWS, and we're used to bouncing best guesses off of it to qualify (e.g. we're constantly keeping linux builds working with the help of EWS)

Reply:
I had just replaced `ENABLE(NOTIFICATIONS)` with `ENABLE(DECLARATIVE_WEB_PUSH)` based on @darinadler's suggestion; I'm certainly fine with the former.

> And on the topic of best guesses - There's a reason we have EWS

No, my entire point here is that I would've happily fixed our build _without_ reverting you if it boiled down to a simple oversight. Changing the constructor of a class which is built unconditionally on all platforms to add a _non-final_ parameter of a type that exists conditionally on some platforms made me feel like your patch was based on expectations that do not hold, and therefore I needed you to be involved in the fix.